contracts/vaults/balancer/internal/pool/TwoTokenPoolUtils.sol

```diff
@@ -258,6 +259,11 @@ library TwoTokenPoolUtils {
             finalPrimaryBalance += primaryPurchased;
         }
 
+        if (strategyContext.vaultState.totalBPTHeld < bptClaim) {
+            strategyContext.vaultState.totalBPTHeld = 0;
```

Comment:
@weitianjie2000 @jeffywu do you think maybe it's a better idea to revert here instead of setting totalBPTHeld to 0? Granted this could get in the way of liquidation, but also this condition should never obtain. So if it does, something is seriously wrong

Reply:
Agree, we should revert in this case.

Reply:
fixed